zebra/router-id.c

```diff
@@ -253,6 +253,34 @@ DEFUN (no_router_id,
 	return CMD_SUCCESS;
 }
 
+DEFUN (show_router_id,
+       show_router_id_cmd,
+       "show router-id [vrf NAME]",
+       "Show the configured router-id\n"
+       VRF_CMD_HELP_STR)
+{
+        int idx_name = 3;
+
+        vrf_id_t vrf_id = VRF_DEFAULT;
+        struct zebra_vrf *zvrf;
+
+        if (argc > 2)
+                VRF_GET_ID(vrf_id, argv[idx_name]->arg, false);
+
+	zvrf = vrf_info_get(vrf_id);
+
+	if ((zvrf != NULL ) && (zvrf->rid_user_assigned.u.prefix4.s_addr)) {
+		if (vrf_id == VRF_DEFAULT)
+			vty_out(vty, "router-id %s\n",
```

Comment:
maybe it would be enough printing just a router-id without "router-id" notation?

Reply:
I wasn't sure which way to go with this one or whether I should keep it consistent with how it appears with "show running-config".  I am happy to change it so only the id is shown.